plugin/completion.py

```diff
@@ -1,22 +1,124 @@
 import sublime
 import sublime_plugin
 
-from .core.protocol import Request, Range, InsertTextFormat
-from .core.settings import settings, client_configs
+from .core.configurations import is_supported_syntax
+from .core.edit import parse_text_edit
 from .core.logging import debug
-from .core.completion import parse_completion_response, format_completion
+from .core.protocol import Request, Range, InsertTextFormat
 from .core.registry import session_for_view, client_from_session, LSPViewEventListener
-from .core.configurations import is_supported_syntax
 from .core.sessions import Session
-from .core.edit import parse_text_edit
-from .core.views import range_to_region
+from .core.settings import settings, client_configs
 from .core.typing import Any, List, Dict, Optional, Union
+from .core.views import range_to_region
 from .core.views import text_document_position_params
-from .core.restore_lines import RestoreLines
+
+
+completion_kinds = {
+    1: (sublime.KIND_ID_MARKUP, "Ξ", "Text"),
+    2: (sublime.KIND_ID_FUNCTION, "λ", "Method"),
+    3: (sublime.KIND_ID_FUNCTION, "λ", "Function"),
+    4: (sublime.KIND_ID_FUNCTION, "c", "Constructor"),
+    5: (sublime.KIND_ID_VARIABLE, "f", "Field"),
+    6: (sublime.KIND_ID_VARIABLE, "v", "Variable"),
+    7: (sublime.KIND_ID_TYPE, "c", "Class"),
+    8: (sublime.KIND_ID_TYPE, "i", "Interface"),
+    9: (sublime.KIND_ID_NAMESPACE, "◪", "Module"),
+    10: (sublime.KIND_ID_VARIABLE, "ρ", "Property"),
+    11: (sublime.KIND_ID_VARIABLE, "u", "Unit"),
+    12: (sublime.KIND_ID_VARIABLE, "ν", "Value"),
+    13: (sublime.KIND_ID_TYPE, "ε", "Enum"),
+    14: (sublime.KIND_ID_KEYWORD, "κ", "Keyword"),
+    15: (sublime.KIND_ID_SNIPPET, "s", "Snippet"),
+    16: (sublime.KIND_ID_AMBIGUOUS, "c", "Color"),
+    17: (sublime.KIND_ID_AMBIGUOUS, "#", "File"),
+    18: (sublime.KIND_ID_AMBIGUOUS, "⇢", "Reference"),
+    19: (sublime.KIND_ID_AMBIGUOUS, "ƒ", "Folder"),
+    20: (sublime.KIND_ID_TYPE, "ε", "EnumMember"),
+    21: (sublime.KIND_ID_VARIABLE, "π", "Constant"),
+    22: (sublime.KIND_ID_TYPE, "s", "Struct"),
+    23: (sublime.KIND_ID_FUNCTION, "e", "Event"),
+    24: (sublime.KIND_ID_KEYWORD, "ο", "Operator"),
+    25: (sublime.KIND_ID_TYPE, "τ", "Type Parameter")
+}
+
+
+class RestoreLines:
```

Comment:
Great that you moved all the code related completions to completions.py.
 
But can you move back only the RestoreLines in the separate file,
just to keep the completions file a bit smaller, 
so other people don't get overwhelmed  by all the code in here?

Reply:
Meh. 300 lines of Python code is not overwhelming.